routers/repo/pull_review.go

```diff
@@ -103,9 +103,29 @@ func SubmitReview(ctx *context.Context, form auth.SubmitReviewForm) {
 	var err error
 
 	reviewType := form.ReviewType()
-	if reviewType == models.ReviewTypeUnknown {
+
+	switch reviewType {
+	case models.ReviewTypeUnknown:
 		ctx.ServerError("GetCurrentReview", fmt.Errorf("unknown ReviewType: %s", form.Type))
 		return
+
+	// can not approve/reject your own PR
+	case models.ReviewTypeApprove, models.ReviewTypeReject:
+
+		if issue.Poster.ID == ctx.User.ID {
+
+			var translated string
+
+			if reviewType == models.ReviewTypeApprove {
+				translated = ctx.Tr("repo.issues.review.self.approval")
+			} else {
+				translated = ctx.Tr("repo.issues.review.self.rejection")
+			}
+
+			ctx.Flash.Error(translated)
+			ctx.Redirect(fmt.Sprintf("%s/pulls/%d", ctx.Repo.RepoLink, issue.Index))
```

Comment:
Please redirect to `%s/pulls/%d/files` instead.

Reply:
The `base/alert` template isn't imported in the template for the page `%s/pulls/%d/files` which is why I redirected back to the main page... Should I just add the alert template in there or just leave it as it is ? ( #4632  redirects to the main PR page though )

Reply:
Add alert to to files template

Reply:
Fixed @JonasFranzDEV and @lafriks